### PR TITLE
PP-8295: Retag sidecar images with xxx-perf

### PR DESCRIPTION
**Pull request by oswaldquek**
This follows the pattern laid out in
https://github.com/alphagov/pay-ci/pull/676.

🟢 jobs can be seen at https://pay-cd.deploy.payments.service.gov.uk/teams/pay-deploy/pipelines/deploy-to-production.

### CODE REVIEW

File: ci/pipelines/deploy-to-production.yml [hunk 1]
```diff
@@ -989,6 +1055,40 @@ jobs:
...
+  - name: retag-stunnel-image-for-test-perf
+    plan:
+      - get: pay-ci
+      - get: stunnel-ecr-registry-prod
+        params:
+          format: oci
+        passed: [deploy-carbon-relay-to-prod]
+        trigger: true
+      - task: parse-perf-release-tag
+        file: pay-ci/ci/tasks/parse-perf-release-tag.yml
```
Comment (katstevens) [inline ci/pipelines/deploy-to-production.yml:1084]: I think we're going to have to do something different for egress, stunnel and carbon relay - the tags will need to be something like `10-egress-perf` (like we have `10-stunnel-release`).

Comment (oswaldquek) [reply ci/pipelines/deploy-to-production.yml:1084]: Yeah you're right

**Comment by oswaldquek** [comment]
Update: we now have a parse-perf-release-tag.js file which [retags egress with 15-egress-release](https://pay-cd.deploy.payments.service.gov.uk/teams/pay-deploy/pipelines/deploy-to-production/jobs/retag-egress-image-for-test-perf/builds/10) and [retags a non-sidecar with the expected](https://pay-cd.deploy.payments.service.gov.uk/teams/pay-deploy/pipelines/deploy-to-production/jobs/retag-adminusers-image-for-test-perf/builds/8).

File: ci/pipelines/deploy-to-production.yml [hunk 2]
```diff
@@ -1188,6 +1322,23 @@ jobs:
           image: frontend-ecr-registry-prod/image.tar
           additional_tags: parse-perf-release-tag/tag
 
+  - name: retag-nginx-forward-proxy-image-for-test-perf
+    plan:
+      - get: pay-ci
+      - get: nginx-forward-proxy-ecr-registry-prod
+        params:
+          format: oci
+        passed: [deploy-frontend-to-prod]
```
Comment (katstevens) [inline ci/pipelines/deploy-to-production.yml:1331]: Do we want to use the `smoke-test-frontend-on-prod` as the check here?

**Review by katstevens — COMMENTED** [review]
Looks good - one question (not blocking, I don't think). Happy to approve once the branch name is updated

File: ci/pipelines/deploy-to-production.yml (hunk 2 above)
Comment (oswaldquek) [reply ci/pipelines/deploy-to-production.yml:1331]: Possibly. At the moment this task is under the [nginx-forward-proxy group](https://github.com/alphagov/pay-ci/blob/master/ci/pipelines/deploy-to-production.yml#L483-L485) which doesn't have the `smoke-test-frontend-on-prod` task. 
It seems like we're using a deploy to frontend to test any new nginx-forward-proxy images.

Comment (oswaldquek) [reply ci/pipelines/deploy-to-production.yml:1331]: Sorry, I might not have been clear. It's not possible to use the `smoke-test-frontend-on-prod` as the check here, unless we move `retag-nginx-forward-proxy-image-for-test-perf` to the [frontend group](https://github.com/alphagov/pay-ci/blob/master/ci/pipelines/deploy-to-production.yml#L429-L434).

Comment (katstevens) [reply ci/pipelines/deploy-to-production.yml:1331]: Ah I see. OK - let's leave as is then.

**Review by katstevens — APPROVED** [review]

**Pull request merged**